pyphare/pyphare/pharesee/run.py

```diff
@@ -89,6 +89,30 @@ def _compute_current(patch):
                 {"name":"Jy", "data":Jy, "centering":centering["Jy"]},
                 {"name":"Jz", "data":Jz,"centering":centering["Jz"]})
 
+
+def _divB2D(Bx,By, xBx, yBy):
+    dxbx = (Bx[1:,:]-Bx[:-1,:])/(xBx[1]-xBx[0])
+    dyby = (By[:,1:]-By[:,:-1])/(yBy[1]-yBy[0])
+    return dxbx + dyby
+
+def _compute_divB(patch):
+
+    if patch.box.ndim ==1:
+        raise ValueError("divB is 0 by construction in 1D")
+
+    elif patch.box.ndim == 2:
+        By  = patch.patch_datas["By"].dataset[:]
+        Bx  = patch.patch_datas["Bx"].dataset[:]
+        xBx = patch.patch_datas["Bx"].x
+        yBy = patch.patch_datas["By"].y
+        divB = _divB2D(Bx,By, xBx, yBy)
+
+        return ({"name":"cell_custom", "data":divB, "centering":"dual"},)
```

Comment:
Why is the centering a single value ? for `ndim=2`, the 3 components of the B fields depends on `x` and `y`, so do the associated div B. I would thus expect `"centering" : ("dual", "dual")` no ?

Reply:
is now resolved in #714. Closing this PR since #714 has this code too.